src/components/organisms/EditProfileModal/EditProfileModal.tsx

```diff
@@ -18,23 +19,26 @@ const EditProfileModal: React.FunctionComponent<PropsType> = ({
   show,
   onHide,
 }) => {
-  const { user, users, profileQuestions } = useSelector((state: any) => ({
-    user: state.user,
-    users: state.firestore.data.users,
+  const { user, profile } = useUser();
+  const { profileQuestions } = useSelector((state: any) => ({
     profileQuestions: state.firestore.data.currentVenue.profile_questions,
   }));
   const onSubmit = async (data: ProfileFormData & QuestionsFormData) => {
+    if (!user) return;
     await updateUserProfile(user.uid, data);
     onHide();
   };
   const defaultValues: any = {
-    partyName: users[user.uid].partyName,
-    pictureUrl: users[user.uid].pictureUrl,
+    partyName: profile?.partyName,
+    pictureUrl: profile?.pictureUrl,
   };
   profileQuestions &&
     profileQuestions.map(
       (question: QuestionType) =>
-        (defaultValues[question.name] = users[user.uid][question.name])
+        (defaultValues[question.name] = profile
+          ? //@ts-ignore
```

Comment:
could you explain why there is a need for this please?

Reply:
profile is a typed object. It cannot be indexed by a string. When I do the typescript refactor, I will type question.name to be a subset of profile keys so that it can safely index the profile.